@xmltree/save.m

```diff
@@ -60,32 +55,37 @@
     end
 end
 
+addstring(fid);
 addstring(sprintf(prolog));
 write_subtree(tree, root(tree), order);
 
-if dofile
-    %keep previous behavior: if nargout is 1, *also* generate string version even if generating file version
+if ~isempty(fid)
     if nargout == 1
-        dofile = false;
-        %addstring(prolog); %2.1 never put prolog in for this case, and note that using fid argument will have erased the prolog string
+        addstring([]);
+        addstring(sprintf(prolog));
```

Comment:
This will not cause it to print any prolog in this case (both filename and output argument), since that variable is still set to the empty string above.  xmltree 2.1 didn't even use the prolog variable in this case.  What do you want it to do?

Reply:
I added it after noticing this discrepancy:

````matlab
>> tree = xmltree('<a>b</a>');
>> xml = save(tree,'file.xml') 

xml =

    '<a>b</a>'

>> type file.xml

<?xml version="1.0" ?>
<a>b</a>
````

Prolog is not displayed only when the input is a file identifier, which is a non documented option so I don't know how much this is used in practice. If needed, the option to print a prolog could be added as an optional argument?

Reply:
Right, I was conflating passing a numeric fid with file output in general, never mind.  I can see the argument for both output methods giving the same result in a single call.